### PR TITLE
[HvA] #47-front-end immutablefive

### PR DESCRIPTION
**Pull request by wooyh2000**
Make sure to set the account on metamask to one of the four accounts in the modules-solodity repo

### CODE REVIEW

File: src/views/Claims.vue [hunk 1]
```diff
@@ -1,37 +1,109 @@
...
     }
 
-    created() {
-        this.$store.dispatch('metamask/checkPreviouslyConnected');
+    /**
+     * Transfers the reward of one token
+     * @param {string} address - The address of the token
+     */
+    async payOneReward(address: string) {
+        await this.contract.methods.withdraw(address).call();
```
Comment (peterpolman) [inline src/views/Claims.vue:95]: @wooyh2000 This one also seems more like a send instead of a call operation. The difference is basically read vs. write.

**Review by Netail — APPROVED** [review]

File: src/views/Claims.vue [hunk 2]
```diff
@@ -1,37 +1,116 @@
...
     }
-    async switchNetwork(networkId: number) {
-        this.$store.dispatch('metamask/requestSwitchNetwork', networkId);
+
+    /**
+     * Update the reward variable and get all unique tokens with their amount and stores it in the tokenAndAmount Object array
+     */
+    async updateReward() {
+        const positionAddress = 0;
+        const positonAmount = 2;
```
Comment (JulianHvA112000) [inline src/views/Claims.vue:69]: Typo, change to 'positionAmount' instead

Comment (wooyh2000) [reply src/views/Claims.vue]: Solved

File: src/views/Claims.vue [hunk 3]
```diff
@@ -1,37 +1,116 @@
...
+        let _amount!: any;
+        this.reward = 0;
+        this.tokenAndAmount = [];
+        try {
+            this.error = false;
+            const response = await this.contract.methods.getRewards(this.account).call();
+            // loop to set all unique tokens to the tokenAndAmount-array with their address and amount
+            for (let i = 0; i < response.length; i++) {
+                // cast to Number, because response returns hexadecimal
+                _amount = Number(response[i][positionAmount]._hex.toString());
```
Comment (JulianHvA112000) [inline src/views/Claims.vue:79]: The rewards are saved as Wei in the smart contract (f.e 0.1745345 = 1745345). Make sure that you use the 'fromWei' web3 function so the amount can be displayed as the proper value (f.e 1745345 = 0.1745345)

**Pull request merged**